applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis.py

```diff
@@ -16,95 +16,106 @@
-        elif (type(project_parameters) == KratosMultiphysics.Parameters): # a Parameters object is provided
-            self.ProjectParameters = project_parameters
-        else:
-            raise Exception("Input is expected to be provided as a Kratos Parameters object or a file name")
-        self._CreateSolver(external_model_part)
+    def __init__(self, model, project_parameters):
+        super(StructuralMechanicsAnalysis, self).__init__(model, project_parameters)
+
+        ## Get echo level and parallel type
+        self.echo_level = self.project_parameters["problem_data"]["echo_level"].GetInt()
+        self.parallel_type = self.project_parameters["problem_data"]["parallel_type"].GetString()
 
-    #### Public functions to run the Analysis ####
-    def Run(self):
-        self.Initialize()
-        self.RunMainTemporalLoop()
-        self.Finalize()
+        ## Import parallel modules if needed
+        if (self.parallel_type == "MPI"):
+            import KratosMultiphysics.mpi as KratosMPI
+            import KratosMultiphysics.MetisApplication as MetisApplication
+            import KratosMultiphysics.TrilinosApplication as TrilinosApplication
+            self.is_printing_rank = (KratosMPI.mpi.rank == 0)
+        else:
+            self.is_printing_rank = True
 
-    def RunMainTemporalLoop(self):
-        while self.time < self.end_time:
-            self.InitializeTimeStep()
-            self.SolveTimeStep()
-            self.FinalizeTimeStep()
+        self._CreateSolver()
 
     def Initialize(self):
+        self.ModifyInitialProperties()
+        self.ModifyInitialGeometry()
         self._ExecuteInitialize()
+        self._SetUpListOfProcesses()
+        ## Processes initialization
+        for process in self.list_of_processes:
+            process.ExecuteInitialize()
+
+        ## Solver initialization
+        self.solver.Initialize()
+
         self._ExecuteBeforeSolutionLoop()
 
-    def InitializeTimeStep(self):
-        self._ExecuteInitializeSolutionStep()
+    def InitializeSolutionStep(self):
+        super(StructuralMechanicsAnalysis, self).InitializeSolutionStep()
+
+        if self.is_printing_rank:
+            KratosMultiphysics.Logger.PrintInfo("STEP: ", self.main_model_part.ProcessInfo[KratosMultiphysics.STEP])
+            KratosMultiphysics.Logger.PrintInfo("TIME: ", self.time)
+        sys.stdout.flush()
 
-    def SolveTimeStep(self):
-        self._SolveSolutionStep()
+        if (self.output_post == True):
+            self.gid_output.ExecuteInitializeSolutionStep()
 
-    def FinalizeTimeStep(self):
-        self._ExecuteFinalizeSolutionStep()
+    def FinalizeSolutionStep(self):
+        super(StructuralMechanicsAnalysis, self).FinalizeSolutionStep()
 
-    def Finalize(self):
-        self._ExecuteFinalize()
+        if (self.output_post == True):
+            self.gid_output.ExecuteFinalizeSolutionStep()
```

Comment:
You can add the output to the standard list of processes and you won't need to redefine this function here (this also ensures that all standard process methods in gid_output_process are called at the right time)